CorsixTH/Lua/ui.lua

```diff
@@ -235,9 +235,13 @@ function UI:setupGlobalKeyHandlers()
   self:addKeyHandler({"alt", "f4"}, self, self.exitApplication)
   self:addKeyHandler({"shift", "f10"}, self, self.resetApp)
 
-  if self.app.config.debug then
-    self:addKeyHandler("f12", self, self.showLuaConsole)
-    self:addKeyHandler({"shift", "d"}, self, self.runDebugScript)
+  self:addOrRemoveDebugModeKeyHandlers()
+end
+
+function UI:connectDebugger()
+  local error_message = TheApp:connectDebugger()
+  if error_message then
+    self.ui:addWindow(UIInformation(self.ui, {error_message}))
```

Comment:
I tried running the debugger without eclipse set up and got a nil pointer here. These references to "self.ui" should just read "self" as you are in the UI class. I made these changes myself and it seemed to show the correct error message then.